packages/web3/types/index.d.ts

```diff
@@ -27,7 +27,7 @@ import { Shh } from 'web3-shh';
 import { Utils } from 'web3-utils';
 
 export default class Web3 {
-    constructor(provider: provider, net?: net.Socket);
+    constructor(provider?: provider, net?: net.Socket);
```

Comment:
Thanks for this, your change will make the typings not as clean you would want, can you please overload them like:

```
constructor(provider: provider);
constructor(provider: provider, net: net.Socket);
constructor(net: net.Socket);
constructor();
```

and then write some tests for them please.